tests/client/test_async_client.py

```diff
@@ -1,12 +1,14 @@
 from datetime import timedelta
+from typing import AsyncGenerator
 
 import pytest
+from uvicorn.main import Server
 
 import httpx
 
 
 @pytest.mark.usefixtures("async_environment")
-async def test_get(server):
+async def test_get(server: Server) -> None:
     url = server.url
```

Comment:
I'm a bit curious about why this works at all — Uvicorn's `Server` doesn't have a `.url` property, but our `TestServer` does. 🤔 I assume it's because Uvicorn doesn't ship type annotations so mypy really just treats `Server` as `Any`? (A `reveal_type(server)` could help confirm it.)

I don't think importing from `conftest` is common practice? If not we might need to move the `TestServer` (and some other items) to `utils.py`. And we might as well rename `TestServer` as `Server`? (Any such changes should  probably go into their own PRs too, right?)

(I realize there are some rough edges to clean up here and there for us to be able to roll out type hints to the tests code more fluently in the future; I think it's okay to do that small bit of work first.)

Reply:
> I'm a bit curious about why this works at all — Uvicorn's Server doesn't have a .url property, but our TestServer does. thinking I assume it's because Uvicorn doesn't ship type annotations so mypy really just treats Server as Any? (A reveal_type(server) could help confirm it.)

Yes, you are right. its type is `Any`.

> I don't think importing from conftest is common practice? If not we might need to move the TestServer (and some other items) to utils.py. And we might as well rename TestServer as Server? (Any such changes should probably go into their own PRs too, right?)

I will do this in separate PR as well

Reply:
What do you think about #991?

I'm thinking adding the `server: Server` annotations everywhere is super cumbersome, and doesn't add a lot of value. We can have mypy type check the inside of our test functions anyway, which will save us a big amount of work. If we go for that approach, then we actually don't really need to move `TestServer` anywhere (since we won't really need to import it for type hints purposes).

Besides, note that I initally submitted #990 but closed it as I think #991 is a better way forward.

Reply:
Thanks for the PR :+1:  . Yes, I agree with you. this change will save our time and we can go faster.